app/models/channel.rb

```diff
@@ -50,4 +53,37 @@ def details_not_changed?
       errors.add(:details, "can't be changed")
     end
   end
+
+  def channel_id
+    channel_type = self.details_type
+    case channel_type
+    when "YoutubeChannelDetails"
+      return self.details.youtube_channel_id
+    when "SiteChannelDetails"
+      return self.details.brave_publisher_id
+    else
+      nil
+    end
+  end
+
+  def promo_enabled?
```

Comment:
I think we should create a model Concern for model ref promo methods
https://stackoverflow.com/questions/14541823/how-to-use-concerns-in-rails-4